.github/CODEOWNERS

```diff
@@ -137,6 +137,7 @@ internal/common/                                                 @open-telemetry
 internal/coreinternal/                                           @open-telemetry/collector-contrib-approvers @open-telemetry/collector-approvers
 internal/datadog/                                                @open-telemetry/collector-contrib-approvers @mx-psi @dineshg13 @liustanley @songy23 @mackjmr @ankitpatel96 @jade-guiton-dd @IbraheemA
 internal/docker/                                                 @open-telemetry/collector-contrib-approvers @jamesmoessis @MovieStoreGuy
+internal/e2e/                                                    @open-telemetry/collector-contrib-approvers @mx-psi @dineshg13 @liustanley @songy23 @mackjmr @ankitpatel96 @jade-guiton-dd @IbraheemA
```

Comment:
I don't think Datadog should own such a general folder 😅 I would rather move this to `internal/datadog/e2e`

Reply:
we can also add more codeowners, those tests seem useful to test general configuration validations.